silkworm/capi/silkworm.cpp

```diff
@@ -473,171 +477,267 @@ class BlockProvider {
     BlockNum max_block_;
 };
 
+class BlockExecutor {
+  public:
+    BlockExecutor(uint64_t chain_id, bool write_receipts, bool write_call_traces, bool write_change_sets, size_t max_batch_size)
+        : write_receipts_{write_receipts},
+          write_call_traces_{write_call_traces},
+          write_change_sets_{write_change_sets},
+          analysis_cache_{kAnalysisCacheSize},
+          state_pool_{},
+          progress_{.start_time = std::chrono::steady_clock::now()},
+          log_time_{progress_.start_time + 20s},
+          max_batch_size_{max_batch_size} {
+        const auto chain_info = kKnownChainConfigs.find(chain_id);
+        chain_config_ = *chain_info;
```

Comment:
We're missing the check on unknown chain id here and it's an unchecked input for both our external endpoints. I suggest to change `BlockExecutor` to receive the chain configuration as `const ChainConfig* chain_config` and restore the check:
```c++
const auto chain_info = kKnownChainConfigs.find(chain_id);
if (!chain_info) {
    return SILKWORM_UNKNOWN_CHAIN_ID;
}
const ChainConfig* chain_config{*chain_info};
```
both in `silkworm_execute_blocks_ephemeral` and `silkworm_execute_blocks_perpetual`.

You could also add a new unit test covering this negative input case.

Reply:
The check already exists in line 601:
```c++
        if (!block_executor.has_known_chain_id()) {
            return SILKWORM_UNKNOWN_CHAIN_ID;
        }
```

I've added unit tests to cover that scenario